teuthology/misc.py

```diff
@@ -818,53 +819,269 @@ def get_wwn_id_map(remote, devs):
+    """
+    Compute the list of free block devices by comparing the list
+    of block devices we have and the one we consider as used
+    """
+    return list(set(available_block_devices) - set(used_block_devices))
+
+
+def get_used_block_devices(remote):
+    """
+    From the actual mount points, let's found out
+    which block devices are used
+    """
+    devs = []
+    r = remote.run(args=['grep', run.Raw('"^/" /proc/mounts')],
+                   stdout=StringIO()
+                   )
+    devices = r.stdout.getvalue().strip().split('\n')
+    for device in devices:
+        used_device = device.split()[0]
+        if used_device not in devs:
+            devs.append(used_device)
 
-    # Remove root device (vm guests) from the disk list
-    for dev in devs:
-        if 'vda' in dev:
-            devs.remove(dev)
-            log.warn("Removing root device: %s from device list" % dev)
+    return devs
 
-    log.debug('devs={d}'.format(d=devs))
 
-    retval = []
-    for dev in devs:
-        try:
-            # FIXME: Split this into multiple calls.
-            remote.run(
-                args=[
-                    # node exists
-                    'stat',
-                    dev,
-                    run.Raw('&&'),
-                    # readable
-                    'sudo', 'dd', 'if=%s' % dev, 'of=/dev/null', 'count=1',
-                    run.Raw('&&'),
-                    # not mounted
-                    run.Raw('!'),
-                    'mount',
-                    run.Raw('|'),
-                    'grep', '-q', dev,
-                ]
-            )
-            retval.append(dev)
-        except CommandFailedError:
-            log.debug("get_scratch_devices: %s is in use" % dev)
-    return retval
+def get_root_device(remote):
+    """
+    Extracting root device
+    """
+    r = remote.run(args=['cat', run.Raw('/proc/cmdline')],
+                   stdout=StringIO()
+                   )
+
+    cmdline = r.stdout.getvalue().strip()
+    matching_line = re.match('.*root=(\S+)\s+.*', cmdline, re.M | re.I)
+    if matching_line:
+        return matching_line.group(1)
+
+    return ""
+
+
+def translate_block_device_path(remote, device):
+    """
+    Extract the cannonized path of block devices
+    """
+    r = remote.run(args=['readlink', run.Raw('-f %s' % device)],
+                   stdout=StringIO()
+                   )
+    return r.stdout.getvalue().strip()
+
+
+def translate_block_devices_path(remote, devices):
+    """
+    Extract the cannonized path of block devices
+    """
+    devs = []
+    for device in devices:
+        device_path = translate_block_device_path(remote, device)
+        if device_path not in devs:
+            devs.append(device_path)
+
+    return devs
+
+
+def expand_dm_devices(remote, devices):
+    """
+    Extract the real device under the DM devices
+    """
+    devs = []
+    for device in devices:
+        if device.startswith("/dev/dm-"):
```

Comment:
@ErwanAliasr1 Thanks for the explanation, multipath/multipathd commands properly deal with devices based on the multipath.conf, you can safely remove the dm logic since I think multipath devices nodes are not available for you to test anywhere in the lab  and also I guess the imaging of new nodes ends up removing any multipath packages from the system even for root devices.